FLT/Mathlib/NumberTheory/Padics/PadicIntegers.lean

```diff
@@ -0,0 +1,56 @@
+import Mathlib.NumberTheory.Padics.PadicIntegers
+import FLT.Mathlib.Algebra.Group.Units.Hom
+
+/-!
+# TODO
+
+* Make `PadicInt.valuation` `ℕ`-valued
+* Rename `Coe.ringHom` to `coeRingHom`
+-/
+
+-- This is cool notation. Should mathlib have it?
+scoped[GroupTheory] notation "[" G ":" H "]" => @AddSubgroup.index G _ H
+
+open Function
+open scoped GroupTheory
+open scoped Pointwise
+
+namespace PadicInt
+variable {p : ℕ} [Fact p.Prime]
+
+@[simp, norm_cast] lemma coe_coeRingHom : ⇑(Coe.ringHom (p := p)) = (↑) := rfl
+
+lemma coe_injective : Injective ((↑) : ℤ_[p] → ℚ_[p]) := Subtype.val_injective
+
+@[simp] lemma coe_inj {x y : ℤ_[p]} : (x : ℚ_[p]) = (y : ℚ_[p]) ↔ x = y := coe_injective.eq_iff
+
+noncomputable instance : Coe ℤ_[p]ˣ ℚ_[p]ˣ where coe := Units.map Coe.ringHom.toMonoidHom
+
+--TODO(javierlcontreras): Waiting for a new mathlib refactor. Z_p valuation should return nat, not int
+--lemma unit_ppown (a : ℤ_[p]) : ∃ u : ℤ_[p]ˣ, u * (p ^ a.valuation) = a := by
+--  sorry
+
+lemma index_ideal (a : ℤ_[p]) : (a • ⊤ : Ideal ℤ_[p]).toAddSubgroup.index = ‖a‖⁻¹ := by
+  sorry
+
+lemma index_ideal_unit (u : ℤ_[p]ˣ) :
+    (u • ⊤ : Ideal ℤ_[p]).toAddSubgroup.index = ‖(u : ℤ_[p])‖⁻¹ := by
+  simp
+  sorry
```

Comment:
They're not just equal, they're both equal to 1. Are you sure you want this lemma? Wouldn't you be better off just proving they were both 1, if you definitely want this?